pkgs/by-name/te/teamspeak3/package.nix

```diff
@@ -0,0 +1,143 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  fetchzip,
+  makeWrapper,
+  makeDesktopItem,
+  glib,
+  qt5,
+  libsForQt5,
+  xkeyboard_config,
+  perl,
+  libcxx,
+  autoPatchelfHook,
+}:
+
+let
+  arch = "amd64";
+
+  pluginsdk = fetchzip {
+    url = "https://files.teamspeak-services.com/releases/sdk/3.3.1/ts_sdk_3.3.1.zip";
+    hash = "sha256-wx4pBZHpFPoNvEe4xYE80KnXGVda9XcX35ho4R8QxrQ=";
+  };
+in
+
+stdenv.mkDerivation rec {
+  pname = "teamspeak-client";
+
+  version = "3.6.2";
+
+  src = fetchurl {
+    url = "https://files.teamspeak-services.com/releases/client/${version}/TeamSpeak3-Client-linux_${arch}-${version}.run";
+    hash = "sha256-WfEQQ4lxoj+QSnAOfdCoEc+Z1Oa5dbo6pFli1DsAZCI=";
+  };
+  # It's an install script
+  dontUnpack = true;
+
+  nativeBuildInputs = [
+    perl # Installer script needs `shasum`
+    qt5.wrapQtAppsHook
+    autoPatchelfHook
+  ];
+
+  buildInputs =
+    [
+      libsForQt5.quazip
+      glib
+      libcxx
+    ]
+    ++ (with qt5; [
+      qtbase
+      qtwebengine
+      qtwebchannel
+      qtwebsockets
+      qtsvg
+    ]);
+
+  # This runs the installer script. If it gets stuck, run it with -x.
+  # If it then gets stuck at something like:
+  #
+  # ++ exec
+  # + PAGER_PATH=
+  #
+  # it's looking for a dependency and didn't find it. Check the script and make
+  # sure the dep is in nativeBuildInputs.
+  buildPhase = ''
+    runHook preBuild
+
+    # Run the installer script non-interactively
+    echo -e '\ny' | PAGER=cat sh -e $src
+
+    # Delete unecessary libraries - these are provided by nixos.
+    pushd TeamSpeak3-Client-linux_amd64/
+
+    # I hate `find`.
+    find . -\( -name '*.so' -or -name '*.so.*' -\) -print0 | xargs -0 rm
```

Comment:
Why `-print0 | xargs -0 rm` vs `-delete`?

Reply:
I invite you to try it; it does not work. `find` is historical garbage and even getting the print0 to work wasn't trivial. I was tempted to just use `fd` here.

Reply:
I tried it; it seems to work?

```
$ rg find pkgs/by-name/te/teamspeak3/package.nix 
60:  # it's looking for a dependency and didn't find it. Check the script and make
77:    find . -\( -name '*.so' -or -name '*.so.*' -\) -delete
$ NIXPKGS_ALLOW_UNFREE=1 nix-build -A teamspeak3 --option builders ''
/nix/store/hx12r49pbb533in5yvz96qq17a4f54vf-teamspeak3-3.6.2
$ ls -R /nix/store/hx12r49pbb533in5yvz96qq17a4f54vf-teamspeak3-3.6.2 | rg '\.so'
$
```

Is there something else that should be being deleted here?

Reply:
I really don't care about this as much as the other thing though, to be honest, so whatever.